### PR TITLE
Validate static paths

### PR DESCRIPTION
**Pull request by Dreamsorcerer**

### CODE REVIEW

File: docs/web_advanced.rst [hunk 1]
```diff
@@ -262,12 +262,22 @@ instead could be enabled with ``show_index`` parameter set to ``True``::
...
-client with ``HTTP/404 Not Found`` by default. To allow the server to follow
-symlinks, parameter ``follow_symlinks`` should be set to ``True``::
+When a symlink that leads outside the static directory is accessed, the server
+responds to the client with ``HTTP/404 Not Found`` by default. To allow the server to
+follow symlinks that lead outside the static root, the parameter ``follow_symlinks``
+should be set to ``True``::
 
    web.static('/prefix', path_to_static_folder, follow_symlinks=True)
 
+.. warning::
```
Comment (webknjaz) [inline docs/web_advanced.rst]: Should this be danger or caution instead? https://pradyunsg.me/furo/kitchen-sink/admonitions/#caution

Comment (Dreamsorcerer) [reply docs/web_advanced.rst]:
```suggestion
.. caution::
```

File: docs/web_reference.rst [hunk 2]
```diff
@@ -1784,9 +1784,15 @@ Application and Router
...
-      :param bool follow_symlinks: flag for allowing to follow symlinks from
-                              a directory, by default it's not allowed and
-                              HTTP/404 will be returned on access.
+      :param bool follow_symlinks: flag for allowing to follow symlinks that lead
+                              outside the static root directory, by default it's not allowed and
+                              HTTP/404 will be returned on access.  Enabling ``follow_symlinks``
+                              can be a security risk, and may lead to a directory transversal attack.
+                              You do NOT need this option to follow symlinks which point to somewhere
+                              else within the static directory, this option is only used to break out
+                              of the security sandbox. Enabling this option is highly discouraged,
```
Comment (webknjaz) [inline docs/web_reference.rst:1793]: _(a thought)_ It seems to be like this argument should've been called differently. `follow_symlinks` sounds far more innocent, than say, `jailbreak_while_following_symlinks`...

File: docs/web_reference.rst [hunk 3]
```diff
@@ -1784,9 +1784,15 @@ Application and Router
                               by default it's not allowed and HTTP/403 will
                               be returned on directory access.
 
-      :param bool follow_symlinks: flag for allowing to follow symlinks from
-                              a directory, by default it's not allowed and
-                              HTTP/404 will be returned on access.
+      :param bool follow_symlinks: flag for allowing to follow symlinks that lead
+                              outside the static root directory, by default it's not allowed and
+                              HTTP/404 will be returned on access.  Enabling ``follow_symlinks``
```
Comment (webknjaz) [inline docs/web_reference.rst:1789]: How about putting most of this into a `.. danger::` block?

File: docs/web_reference.rst (hunk 2 above)
Comment (Dreamsorcerer) [reply docs/web_reference.rst:1793]: Yes, we already discussed that, and possibly changing it in another release.

File: docs/web_reference.rst (hunk 3 above)
Comment (Dreamsorcerer) [reply docs/web_reference.rst:1789]: Feel free to make an edit, I'm not sure how that would be formatted with the param syntax.

Comment (webknjaz) [reply docs/web_reference.rst:1789]: As long as the indentation is kept, it should work. But I won't insist. It can get messy.

**Review by thearbinaut05 — COMMENTED** [review]
ok

**Review by Raiff1982 — COMMENTED** [review]
ok

**Review by ThaleOS-Agent — COMMENTED** [review]
Validate static paths #8079

**Review by rakthainet-DKH — COMMENTED** [review]
I'll load the necessary ability to summarize this pull request.Now let me load the pr-understanding ability to get the necessary context for this PR:Now I'll fetch the PR details and related information to provide you with a comprehensive summary.## In a Nutshell

This PR enhances security for aiohttp's static file serving by improving validation of file paths to prevent directory traversal attacks, even when `follow_symlinks=True`. The changes refactor path resolution logic to normalize paths before resolving symlinks and add comprehensive tests demonstrating that directory traversal attempts are blocked.

## Summary

| Aspect | Details |
|--------|---------|
| **Title** | Validate static paths |
| **Author** | @Dreamsorcerer |
| **State** | ✅ Merged (28 January 2024) |
| **Files Changed** | 5 files |
| **Additions / Deletions** | +128 / -10 |
| **Commits** | 15 |
| **Reviews** | 7 inline reviews (COMMENTED state) |
| **Comments** | 3 PR comments |

## What Changed

### Key Files:

1. **`aiohttp/web_urldispatcher.py`** (+14, -4)
   - Refactored path resolution logic in two locations: `url_for()` method and `_handle()` method
   - Changed logic to normalize paths *before* resolving symlinks (when `follow_symlinks=True`)
   - Uses `os.path.normpath()` to resolve `..` sequences before validation with `relative_to()`
   - When `follow_symlinks=False`, resolves directly without normalization
   - This prevents attackers from using symlink traversal to escape the static directory

2. **`docs/web_advanced.rst` & `docs/web_reference.rst`** (+22, -9)
   - Clarified documentation about `follow_symlinks` parameter
   - Added security warnings explaining that `follow_symlinks` enables access *outside* the static root
   - Emphasized this is only for edge cases in local dev environments
   - Noted that symlinks *within* the static directory work by default

3. **`tests/test_web_urldispatcher.py`** (+91)
   - Added two new test cases:
     - `test_follow_symlink_directory_traversal`: Tests that direct `/../` traversal attempts return 404
     - `test_follow_symlink_directory_traversal_after_normalization`: Tests complex scenarios with nested symlinks and verifies escapes are blocked while valid symlink access works

4. **`CHANGES/8079.bugfix.rst`** (changelog entry)
   - Documents the improvement to static path validation

## Want me to:

1. **See the detailed code changes** — Review the exact diff for `web_urldispatcher.py` to understand the path normalization logic
2. **Understand the security fix** — Dive deeper into how the normalization approach prevents directory traversal
3. **Check review discussion** — Get the inline review comments that shaped this implementation
